pkg/config/lvmd/lvmd.go

```diff
@@ -0,0 +1,104 @@
+// DeviceClass type definition and validation is taken from:
+// https://github.com/red-hat-storage/topolvm/blob/release-4.10/lvmd/device_class_manager.go
+package lvmd
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultSpareGB  = 10
+	defaultSockName = "/run/lvmd/lvmd.socket"
+)
+
+var qualifiedNameRegexp = regexp.MustCompile("^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$")
+
+// This regexp is used to check StripeSize format
+var stripeSizeRegexp = regexp.MustCompile("(?i)^([0-9]*)(k|m|g|t|p|e|b|s)?$")
+
+// DeviceClass maps between device-classes and volume groups.
+type DeviceClass struct {
+	// Name for the device-class name
+	Name string `yaml:"name"`
+	// Volume group name for the device-class
+	VolumeGroup string `yaml:"volume-group"`
+	// Default is a flag to indicate whether the device-class is the default
+	Default bool `yaml:"default"`
+	// SpareGB is storage capacity in GiB to be spared
+	SpareGB uint `yaml:"spare-gb"`
+	// Stripe is the number of stripes in the logical volume
+	Stripe uint `yaml:"stripe,omitempty"`
+	// StripeSize is the amount of data that is written to one device before moving to the next device
+	StripeSize string `yaml:"stripe-size,omitempty"`
+}
+
+type CSIPluginConfig struct {
+	DeviceClasses []*DeviceClass `yaml:"deviceClasses"`
+}
+
+// lvmdConfig appends the SocketName to the user facing Config. The default value defined in this package represent the
+// mount location in the lvmd pod.
+type lvmdConfig struct {
+	DeviceClasses []*DeviceClass `yaml:"device-classes"`
+	SocketName    string         `yaml:"socket-name"`
+}
+
+func (c *lvmdConfig) String() string {
+	out, err := yaml.Marshal(c)
+	if err != nil {
+		klog.Fatalf("processing lvmd lvmdConfig: %v", err)
+	}
+	return fmt.Sprintf(`%q`, string(out))
+}
+
+func NewLVMDConfigForDeviceClasses(d []*DeviceClass) *lvmdConfig {
+	return &lvmdConfig{
+		SocketName:    defaultSockName,
+		DeviceClasses: d,
+	}
+}
+
+func ValidateDeviceClasses(deviceClasses []*DeviceClass) error {
```

Comment:
Would it make sense to use the validation in the vendored code instead of reproducing it? Or does that drag in a bunch of other dependencies?